pysradb/utils.py

```diff
@@ -4,20 +4,77 @@
 import ntpath
 import os
 import shlex
-import struct
 import subprocess
-import sys
 import urllib.request as urllib_request
 import warnings
 
+import requests
+from requests import HTTPError
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
 from tqdm.autonotebook import tqdm
 
+from .exceptions import IncorrectFieldException
+
 warnings.simplefilter(action="ignore", category=FutureWarning)
 
 
 tqdm.pandas()
 
 
+def scientific_name_to_taxid(name):
+    """Converts a scientific name to its corresponding taxonomy ID.
+
+    Parameters
+    ----------
+    name: str
+        Scientific name of interest.
+
+    Returns
+    -------
+    taxid: str
+        Taxonomy Id of the Scientific name.
+
+    Raises
+    ------
+    IncorrectFieldException
+        If the scientific name cannot be found.
+
+    """
+
+    r = requests.get(
+        "https://www.ebi.ac.uk/ena/data/taxonomy/v1/taxon/scientific-name/" + name,
+        timeout=5,
+    )
+    try:
+        r.raise_for_status()
+    except HTTPError:
+        raise IncorrectFieldException(f"Unknown scientific name: {name}")
```

Comment:
Could you be more precise here ? `HTTPError`s could have many different cases. Is there a given http status code or even a message that indicates `Unknown scientific name` ?